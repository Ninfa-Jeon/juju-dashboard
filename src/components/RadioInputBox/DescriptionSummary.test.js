import { mount } from "enzyme";

import DescriptionSummary from "./DescriptionSummary";

describe("DescriptionSummary", () => {
  it("returns a details component if the description is longer than 30 characters", () => {
    const description =
      "The number of coding chunks, i.e. the number of additional chunks computed by the encoding functions. If there are 2 coding chunks, it means 2 OSDs can be out without losing data. ";
    const wrapper = mount(<DescriptionSummary description={description} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("returns just the text if the description is shorter than 30 characters", () => {
    const description = "The name of the profile";
    const wrapper = mount(<DescriptionSummary description={description} />);
    expect(wrapper.children().equals(<span>{description}</span>)).toBe(true);
  });

  it("returns null if no description is provided", () => {
    const wrapper = mount(<DescriptionSummary description={undefined} />);
    expect(wrapper.children().length).toBe(0);
  });
});
