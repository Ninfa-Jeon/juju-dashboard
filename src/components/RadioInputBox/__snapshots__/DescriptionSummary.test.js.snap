// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionSummary returns a details component if the description is longer than 30 characters 1`] = `
<DescriptionSummary
  description="The number of coding chunks, i.e. the number of additional chunks computed by the encoding functions. If there are 2 coding chunks, it means 2 OSDs can be out without losing data. "
>
  <details
    className="radio-input-box__details"
  >
    <summary
      className="radio-input-box__summary"
    >
      <span
        className="radio-input-box__summary-description"
      >
        The number of coding chunks, i.e. the number of additional chunks computed by the encoding functions. If there are 2 coding chunks, it means 2 OSDs can be out without losing data. 
      </span>
      Â 
    </summary>
    <span
      className="radio-input-box__details-description"
    >
      The number of coding chunks, i.e. the number of additional chunks computed by the encoding functions. If there are 2 coding chunks, it means 2 OSDs can be out without losing data. 
    </span>
  </details>
</DescriptionSummary>
`;
